### PR TITLE
Update CHANGELOG.md for `0.52.0` release, add scripts/ instructions for ASF releases

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/apache/datafusion-sqlparser-rs/issues/1423

This is the first release that will be officially voted / approved by the Apache voting process and this PR
1. Updates the version to 0.52.0
2. Adds the necessary scripts for making a release (based off what is in the datafusion repo)
3. Adds instructions in `dev/release/README.md`

Draft until: 
- [x] https://github.com/apache/datafusion-sqlparser-rs/pull/1502
- [x] https://github.com/apache/datafusion-sqlparser-rs/pull/1503
- [x] Port the tarball creation scripts
- [x] Test it out for the first time

Following the directions results in this:

```shell
(venv) andrewlamb@Andrews-MacBook-Pro-2:~/Software/sqlparser-rs$ GITHUB_TOKEN=FOO ./dev/release/create-tarball.sh 0.52.0 2
Attempting to create  from tag v0.52.0-rc2
Draft email for dev@datafusion.apache.org mailing list

---------------------------------------------------------
To: dev@datafusion.apache.org
Subject: [VOTE] Release Apache DataFusion sqlparser-rs 0.52.0 RC2
Hi,

I would like to propose a release of Apache DataFusion sqlparser-rs version 0.52.0.

This release candidate is based on commit: 6b491c659ab8a350ee333f489251c7ce446af729 [1]
The proposed release tarball and signatures are hosted at [2].
The changelog is located at [3].

Please download, verify checksums and signatures, run the unit tests, and vote
on the release. The vote will be open for at least 72 hours.

Only votes from PMC members are binding, but all members of the community are
encouraged to test the release and vote with "(non-binding)".

The standard verification procedure is documented at https://github.com/apache/datafusion-sqlparser-rs/blob/main/dev/release/README.md#verifying-release-candidates.

[ ] +1 Release this as Apache DataFusion sqlparser-rs 0.52.0
[ ] +0
[ ] -1 Do not release this as Apache DataFusion sqlparser-rs 0.52.0 because...

Here is my vote:

+1

[1]: https://github.com/apache/datafusion-sqlparser-rs/tree/6b491c659ab8a350ee333f489251c7ce446af729
[2]: https://dist.apache.org/repos/dist/dev/datafusion/apache-datafusion-sqlparser-rs-0.52.0-rc2
[3]: https://github.com/apache/datafusion-sqlparser-rs/blob/6b491c659ab8a350ee333f489251c7ce446af729/CHANGELOG.md
```

You can actually check out a release candidate I made here:
* https://github.com/apache/datafusion-sqlparser-rs/tree/6b491c659ab8a350ee333f489251c7ce446af729
*  https://dist.apache.org/repos/dist/dev/datafusion/apache-datafusion-sqlparser-rs-0.52.0-rc2
* Changelog: https://github.com/apache/datafusion-sqlparser-rs/blob/6b491c659ab8a350ee333f489251c7ce446af729/CHANGELOG.md

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,12 +18,12 @@
 [package]
 name = "sqlparser"
 description = "Extensible SQL Lexer and Parser with support for ANSI SQL:2011"
-version = "0.51.0"
-authors = ["Andy Grove <andygrove73@gmail.com>"]
-homepage = "https://github.com/sqlparser-rs/sqlparser-rs"
+version = "0.52.0"
```
Comment (alamb) [inline Cargo.toml:21]: Updated to reflect apache author (sorry @andygrove !)

File: dev/release/check-rat-report.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python
```
Comment (alamb) [inline dev/release/check-rat-report.py:1]: copied from datafusion without modification

File: dev/release/verify-release-candidate.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
```
Comment (alamb) [inline dev/release/verify-release-candidate.sh:1]: I tested this locally and it seems to work

**Review by iffyio — APPROVED** [review]
LGTM!

**Comment by alamb** [comment]
Thanks @iffyio  -- Let's get this in and give making a release a try!

**Comment by alamb** [comment]
🚀

**Pull request merged**